classes/war/WarAction.php

```diff
@@ -193,55 +207,53 @@ public function progressActiveOperation(): string
+                case self::WAR_ACTION_REINFORCE:
                     switch ($location_target['type']) {
                         case 'castle':
-                            $location_target['max_health'] = WarManager::BASE_CASTLE_HEALTH;
+                            $location_target['max_health'] = WarManager::getLocationMaxHealth($this->system, $location_target);
                             break;
                         case 'village':
-                            $location_target['max_health'] = WarManager::BASE_VILLAGE_HEALTH;
+                            $location_target['max_health'] = WarManager::getLocationMaxHealth($this->system, $location_target);
                             break;
                         default:
                             break;
                     }
                     if ($location_target['health'] < $location_target['max_health']) {
                         $player_heal = floor($this->user->level / 2);
+                        $player_heal *= 1 + ($this->user->village->active_upgrade_effects[VillageUpgradeConfig::UPGRADE_EFFECT_REINFORCE_HEAL] / 100);
                         $new_health = $location_target['health'] + $player_heal;
                         $actual_heal = min($new_health, $location_target['max_health']) - $location_target['health'];
                         $location_target['health'] = $location_target['health'] + $actual_heal;
                         $message .= "Restored " . $actual_heal . " health to " . $location_target['name'] . "!";
                         WarLogManager::logAction($this->system, $this->user, $actual_heal, WarLogManager::WAR_LOG_DAMAGE_HEALED, $this->target_village);
-                        $this->system->db->query("UPDATE `region_locations` SET `health` = {$location_target['health']} WHERE `id` = {$this->target_id}");
-                    } else {
-                        $message .= "Target is already at max health!";
+                        $this->system->db->query("UPDATE `region_locations` SET `health` = {$location_target['health']} WHERE `region_location_id` = {$this->target_id}");
                     }
                     break;
-                case self::OPERATION_RAID:
+                case self::WAR_ACTION_RAID:
                     if ($location_target['health'] > 0) {
                         //$player_damage = max($this->user->level - $location_target['defense'], 0);
                         $defense_reduction = min($location_target['defense'] / 100, 1);
-                        $player_damage = intval($this->user->level * (1 - $defense_reduction));
+                        $player_damage = $this->user->level;
+                        $player_damage *= 1 + ($this->user->village->active_upgrade_effects[VillageUpgradeConfig::UPGRADE_EFFECT_RAID_DAMAGE] / 100);
+                        $player_damage = intval($player_damage * (1 - $defense_reduction));
                         $player_damage = max($player_damage, 0);
```

Comment:
This switch statement is getting a little long, we could consider pulling some or all of the logic for each operation into a method.

For the "some of the logic" approach it could be something like this
```php
if($location_target['health'] > 0) {
  $player_damage = $this->getPlayerRaidDamage($location_target);
  // ...
```